hackathon_site/dashboard/frontend/src/components/orders/OrdersTable/OrdersTable.tsx

```diff
@@ -0,0 +1,164 @@
+import CancelledIcon from "../../../assets/images/icons/statusIcons/cancelled-status.svg";
+import ReturnedIcon from "../../../assets/images/icons/statusIcons/checkout-status.svg";
+import PendingIcon from "../../../assets/images/icons/statusIcons/pending-status.svg";
+import InProgressIcon from "../../../assets/images/icons/statusIcons/inprogress-status.svg";
+
+interface OrdersTableProps {
+    ordersData: Order[];
+}
+
+interface OrderStateIcon {
+    status: string;
+}
+
+const OrderStateIcon = ({ status }: OrderStateIcon) => {
+    const filterState = status.replace(/\s+/g, "");
+    console.log(filterState);
+
+    return (
+        <div className={styles.container}>
+            <div className={`${styles.container} order-${filterState}-icon`}>
+                {/* <img src={filterState + "Icon"} />
+                <img src={`../../../assets/images/icons/statusIcons/${filterState}-status.svg`} /> */}
+                {status === "Submitted" ? (
+                    <img
+                        src={SubmittedIcon}
+                        className={styles.statusIcon}
+                        alt={`${status} icon`}
+                    />
+                ) : status === "Ready for Pickup" ? (
+                    <img
+                        src={ReadyForPickupIcon}
+                        className={styles.statusIcon}
+                        alt={`${status} icon`}
+                    />
+                ) : status === "Picked Up" ? (
+                    <img
+                        src={PickedUpIcon}
+                        className={styles.statusIcon}
+                        alt={`${status} icon`}
+                    />
+                ) : status === "Cancelled" ? (
+                    <img
+                        src={CancelledIcon}
+                        className={styles.statusIcon}
+                        alt={`${status} icon`}
+                    />
+                ) : status === "Returned" ? (
+                    <img
+                        src={ReturnedIcon}
+                        className={styles.statusIcon}
+                        alt={`${status} icon`}
+                    />
+                ) : status === "Pending" ? (
+                    <img
+                        src={PendingIcon}
+                        className={styles.statusIcon}
+                        alt={`${status} icon`}
+                    />
+                ) : (
+                    <img
+                        src={InProgressIcon}
+                        className={styles.statusIcon}
+                        alt={`${status} icon`}
+                    />
+                )}
+                {status}
+            </div>
+        </div>
+    );
+};
+
+const handleEvent: GridEventListener<"rowClick"> = (
+    params, // GridRowParams
+    event, // MuiEvent<React.MouseEvent<HTMLElement>>
+    details // GridCallbackDetails
+) => {
+    console.log("clicked", params.row);
```

Comment:
this handleEvent can be later used to open the page of the specific Order and activates on double click.
does double click sound ok? or single click also works

Reply:
Yeah I like the double click feature